src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -1,11 +1,12 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
 /**
  * Parameterize this service and add its implementation in a separate class.
  */
-public interface MachineService {
+public interface MachineService<T extends Machine> {
```

Comment:
We don't need to set any bounding here, because `MachineServiceImpl` parametrizes this interface with `Machine` type in its signature. If this bounding is set, our interface is depending on the exact implementation (which is working only with the `Machine` hierarchy), therefore, this interface is not reusable in any other part of the program.
```suggestion
public interface MachineService<T> {
```